--- a/lib/DirectoryWatcher.js
+++ b/lib/DirectoryWatcher.js
@@ -855,7 +855,9 @@
 						const missingWatchers = new Map(this.watchers);
 						missingWatchers.delete(withoutCase(this.path));
 						for (const item of itemPaths) {
-							missingWatchers.delete(withoutCase(item));
+							if (this.files.has(item) || this.directories.has(item)) {
+								missingWatchers.delete(withoutCase(item));
+							}
 						}
 						for (const watchers of missingWatchers.values()) {
 							for (const watcher of watchers) {
